### PR TITLE
Add ovn-upgrade presubmits for MCO PRs

### PR DESCRIPTION
**Pull request by jluhrsen**
JIRA: https://issues.redhat.com/browse/SDN-2761

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by jluhrsen** [comment]
@kikisdeliveryservice, the ovnk team was hoping to get these important e2e-ovn-upgrade jobs as presubmits in MCO to help us stave off regressions.

@trozet ^^

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -1,5 +1,128 @@
...
+    cluster: build03
+    context: ci/prow/4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade
+    decorate: true
+    labels:
+      ci-operator.openshift.io/cloud: aws
+      ci-operator.openshift.io/cloud-cluster-profile: aws
+      ci-operator.openshift.io/variant: 4.11-upgrade-from-stable-4.10
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-master-4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:17]: add in all optional: true

**Review by kikisdeliveryservice — CHANGES_REQUESTED** [review]
please make the jobs optional

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 1 above)
Comment (jluhrsen) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:17]: done :)

Comment (kikisdeliveryservice) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:17]: thanks

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -1,5 +1,129 @@
...
+    - ^master-
+    cluster: build03
+    context: ci/prow/4.11-upgrade-from-stable-4.10-images
+    decorate: true
+    labels:
+      ci-operator.openshift.io/variant: 4.11-upgrade-from-stable-4.10
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-machine-config-operator-master-4.11-upgrade-from-stable-4.10-images
+    rerun_command: /test 4.11-upgrade-from-stable-4.10-images
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:91]: Sorry but this is new right? optional:true as well?

Comment (jluhrsen) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:91]: this is the -images job which, if I understand correctly, comes from the below config in the job config .yaml:

```
images:
- from: base
  inputs:
    ocp_4.10_cli:
      as:
      - registry.ci.openshift.org/ocp/4.10:cli
    ocp_builder_rhel-8-golang-1.17-openshift-4.10:
      as:
      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10
  to: machine-config-operator
```

it's a 6m job that builds a release image from the PR. [here's](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/26855/rehearse-26855-pull-ci-openshift-machine-config-operator-master-4.11-upgrade-from-stable-4.10-images/1501998936498376704) what it looks like. Do we want to get rid of this automatic?

**Comment by kikisdeliveryservice** [comment]
Q: is this something that should be run on every PR? Do we have examples where non-ovn touching MCO PRs cause regressions? Could this be be just invoked manually?

**Comment by jluhrsen** [comment]
> Q: is this something that should be run on every PR? Do we have examples where non-ovn touching MCO PRs cause regressions? Could this be be just invoked manually?

well, I think as OVN moving forward as the default CNI it would make sense to run on every PR and maybe back-off the
openshift-sdn version of these upgrade jobs to not run by default. It might even be a good reason to make these ovn
upgrade jobs not be optional. Their health has been steadily improving and I expect them to pass around 50% of the
time.

**Comment by kikisdeliveryservice** [comment]
> Their health has been steadily improving and I expect them to pass around 50% of the
> time.


If they only pass 50% of the time what signal can they give if run on every PR? They certainly can't be non-optional and blocking in that case either.The MCO has drastically reduced the number of always run in favor of manually running where applicable along with encouraging periodics. There were many tests that failed and didn't give PR authors any signal and end up ignored.

**Comment by jluhrsen** [comment]
> > Their health has been steadily improving and I expect them to pass around 50% of the
> > time.
> 
> If they only pass 50% of the time what signal can they give if run on every PR? They certainly can't be non-optional and blocking in that case either.The MCO has drastically reduced the number of always run in favor of manually running where applicable along with encouraging periodics. There were many tests that failed and didn't give PR authors any signal and end up ignored.

fair enough. all jobs are now `always_run: false` as well as `optional: true`

What we noticed in ovnk is that jobs that never ran would go unnoticed and folks would neglect to run them at all. issuing
/retest has become so automatic that a job with 50% pass rate will eventually pass if it's not a problem with the PR. If
it never passes, then we can dig in to the problem of the PR.

In no way am I advocating for this kind of process. I prefer 100% passing and stable jobs to be running and required.
that doesn't seem to be the normal workflow around openshift, from what I can see :)

**Comment by jluhrsen** [comment]
@kikisdeliveryservice, is this good to go now?

**Review by kikisdeliveryservice — APPROVED** [review]

**Pull request merged**